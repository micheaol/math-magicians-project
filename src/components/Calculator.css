.wrapper {
  width: 50%;
  height: 440px;
  margin-left: 10%;
  padding: 10px;
  background-color: #485461;
  background-image: linear-gradient(315deg, #485461 0%, #28313b 74%);
}

.main-wrapper {
  display: flex;
  justify-content: center;
  width: 100%;
  margin-top: 5%;
}
