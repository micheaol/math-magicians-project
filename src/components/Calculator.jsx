/* eslint-disable linebreak-style */
/* eslint-disable arrow-body-style */
import './Calculator.css';
import React, { useState } from 'react';
import ButtonWraper from './ButtonWraper';
import Screen from './Screen';
import Button from './Button';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: null,
    next: null,
    operation: null,
  });

  const { total, next, operation } = state;

  const setNewState = (value) => {
    const newState = calculate(state, value);
    setState(newState);
  };

  const handleClick = (e) => {
    const buttonName = e.target.textContent;
    setNewState(buttonName);
  };

  const btnValues = [
    ['AC', '+/-', '%', 'รท'],
    [7, 8, 9, 'x'],
    [4, 5, 6, '-'],
    [1, 2, 3, '+'],
    [0, '.', '='],
  ];
  return (
    <div className="wrapper">
      <Screen value={operation == null
        ? total || next || '0'
        : `${total} ${operation} ${next == null ? ' ' : next}`}
      />
      <ButtonWraper>
        {
            btnValues.flat().map((btn) => {
              return (
                <Button
                  key={btn}
                  className={btn === '=' ? 'equals' : ''}
                  value={btn}
                  onClick={
                    handleClick
                  }
                />
              );
            })
          }
      </ButtonWraper>
    </div>
  );
};

export default Calculator;
