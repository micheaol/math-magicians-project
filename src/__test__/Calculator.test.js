import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

afterEach(() => {
  cleanup();
});
describe('render calculator component to the screen', () => {
  test('render calculator to the screen', () => {
    render(<Calculator />);
    const calculatorElement = screen.getByTestId('calculator');
    expect(calculatorElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
