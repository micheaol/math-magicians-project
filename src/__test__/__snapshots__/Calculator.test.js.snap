// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render calculator component to the screen matches snapshot 1`] = `
<div
  className="main-wrapper"
  data-testid="calculator"
>
  <h1>
    Lets do some math!
  </h1>
  <div
    className="wrapper"
  >
    <div
      className="screen"
    >
      <h1>
        0
      </h1>
    </div>
    <div
      className="buttonBox"
    >
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        AC
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        +/-
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        %
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        รท
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        7
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        8
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        9
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        x
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        4
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        5
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        6
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        -
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        1
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        2
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        3
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        +
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        0
      </button>
      <button
        className=""
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        .
      </button>
      <button
        className="equals"
        data-testid="button"
        onClick={[Function]}
        type="submit"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
